src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/endpoints/JurisdictionEndpoint.java

```diff
@@ -0,0 +1,93 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.endpoints;
+
+import feign.FeignException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.actuate.endpoint.annotation.Endpoint;
+import org.springframework.boot.actuate.endpoint.annotation.ReadOperation;
+import org.springframework.boot.actuate.endpoint.annotation.Selector;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.Credential;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.JurisdictionToUserMapping;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.NoUserConfiguredException;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.util.Map.Entry;
+
+@Component
+@Endpoint(id = "jurisdictions")
+@EnableConfigurationProperties(JurisdictionToUserMapping.class)
+public class JurisdictionEndpoint {
+
+    private static final Logger log = LoggerFactory.getLogger(JurisdictionEndpoint.class);
+
+    private final Map<String, Credential> jurisdictions;
+    private final Function<String, Credential> credentialProvider;
+    private final IdamClient idamClient;
+
+    public JurisdictionEndpoint(
+        JurisdictionToUserMapping mapping,
+        IdamClient idamClient
+    ) {
+        jurisdictions = mapping.getUsers();
+        credentialProvider = mapping::getUser;
+        this.idamClient = idamClient;
+    }
+
+    @ReadOperation
+    public Map<String, HttpStatus> jurisdictions() {
+        return jurisdictions
+            .entrySet()
+            .stream()
+            .collect(Collectors.toMap(
+                Entry::getKey,
+                entry -> checkCredentials(entry.getKey(), entry.getValue())
+            ));
+    }
+
+    @ReadOperation
+    public HttpStatus jurisdiction(@Selector String jurisdiction) {
+        try {
+            return checkCredentials(jurisdiction, credentialProvider.apply(jurisdiction));
+        } catch (NoUserConfiguredException exception) {
```

Comment:
`checkCredentials` never throws an exception (it has a `catch (Exception exception)` inside), therefore this code is never reached.

Reply:
also, missing test I guess...

Reply:
You are correct `checkCredentials` doesn't throw an exception. But `credentialProvider.apply` does though:

```java
// assignment
credentialProvider = mapping::getUser;

// implementation
public Credential getUser(String jurisdiction) {
    return users.computeIfAbsent(jurisdiction.toLowerCase(), this::throwNotFound);
}

private Credential throwNotFound(String jurisdiction) {
    throw new NoUserConfiguredException(jurisdiction);
}
```

And the test is called `Should respond UNAUTHORISED for specific jurisdiction which does not exist` which verifies `idamClient` is never called.

To summarise: `checkCredentials` function should not throw an exception but always provide some status so the `/jurisdictions` endpoint always responds with success. Furthermore it parses all configured jurisdictions so you are safe from the one pointed out here ^. On the other hand `/jurisdictions/{value}` is free text and therefore can throw extra exception of jurisdiction not being present in config.

P.S. In case missed, coverage reports all lines/branches are satisfied:
> The diff coverage is `100%`.